flight/Modules/UAVOMSPBridge/UAVOMSPBridge.c

```diff
@@ -333,42 +333,50 @@ static void _msp_send_altitude(struct msp_bridge *m)
 	msp_send(m, MSP_ALTITUDE, data.buf, sizeof(data));
 }
 
+static uint16_t _msp_scale_rc(ActuatorSettingsData actuatorSettings,
+			      ManualControlCommandData manualState,
+			      int channel_num)
+{
+	uint16_t val = manualState.Channel[channel_num];
+	uint16_t min = actuatorSettings.ChannelMin[channel_num];
+	uint16_t max = actuatorSettings.ChannelMax[channel_num];
+	uint16_t idle = actuatorSettings.ChannelNeutral[channel_num];
+
+	// Min and max may be reversed.  If the value is outside of
+	// the given range, return the neutral value.
+	uint16_t lower = min < max ? min : max;
+	uint16_t higher = max > min ? max : min;
+	if (val > lower || val < higher) {
+		return idle;
```

Comment:
No, I ran into an issue where during failsafe the values are returned out of range.  In the specific case of throttle, this causes the OSD to learn that the range is wider than it'd previously seen and instead of going 0-100%, it goes 0-1%.

I don't know if there's a better way to detect this, but @mlyle suggested just looking at the value to ensure it's in range.